lib/xrp/XrpRpc.js

```diff
@@ -0,0 +1,248 @@
+const RippleAPI = require('ripple-lib').RippleAPI;
+const promptly = require('promptly');
+const util = require('util');
+const EventEmitter = require('events');
+
+const passwordPromptAsync = util.promisify(promptly.password);
+
+class XrpRpc {
+  constructor(config) {
+    this.config = config;
+    const {
+      rpcPort,
+      host,
+      protocol,
+      address
+    } = config;
+    const connectionString = `${protocol}://${host}:${rpcPort}`;
+    this.rpc = new RippleAPI({
+      server: connectionString
+    });
+    this.address = address;
+    this.emitter = new EventEmitter();
+  }
+
+  async asyncCall(method, args) {
+    return new Promise((resolve) => {
+      this.rpc.connect().then(() => {
+        return this.rpc[method](args).then((result) => {
+          this.rpc.disconnect().then(() => {
+            return resolve(result);
+          });
+        });
+      });
+    });
+  }
+
+  async asyncRequest(method, args) {
+    return new Promise((resolve) => {
+      this.rpc.connect().then(() => {
+        return this.rpc.request(method, args).then((result) => {
+          this.rpc.disconnect().then(() => {
+            return resolve(result);
+          });
+        });
+      });
+    });
+  }
+
+  async unlockAndSendToAddress({ address, amount, secret }) {
+    if (secret === undefined) {
+      secret = await passwordPromptAsync('> ');
+    }
+    console.warn('Unlocking for a single transaction.');
+    return await this.sendToAddress({ address, amount, secret });
+  }
+
+  async sendToAddress({ address, amount, passphrase, tag, invoiceID, secret, alreadyConnected}) {
+    return this.signTransaction({address, amount, passphrase, tag, invoiceID, secret}).then((signedTx) => {
+      return this.sendRawTransaction({ signedTx, alreadyConnected }).then((txHash) => {
+        return txHash;
+      });
+    });
+  }
+
+  async getRawTransaction({ address, amount, tag, invoiceID, sourceAddress }) {
+    const payment = {
+      source: {
+        address: sourceAddress,
+        tag: tag || undefined,
+        maxAmount: {
+          value: amount.toString(),
+          currency: this.config.currency
+        }
+      },
+      destination: {
+        address: address,
+        tag: tag || undefined,
+        amount: {
+          value: amount.toString(),
+          currency: this.config.currency
+        }
+      },
+      invoiceID: invoiceID || undefined
+    };
+    await this.rpc.connect();
+    let prepared = await this.rpc.preparePayment(sourceAddress, payment);
```

Comment:
should we be using asyncCall here to do the connection and disconnection

Reply:
Removed these since method does not use alreadyConnected. Can be refactored in later PR if batching is needed.